src/ripple/protocol/STXChainBridge.h

```diff
@@ -0,0 +1,238 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2022 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_PROTOCOL_STXCHAINBRIDGE_H_INCLUDED
+#define RIPPLE_PROTOCOL_STXCHAINBRIDGE_H_INCLUDED
+
+#include <ripple/protocol/STAccount.h>
+#include <ripple/protocol/STBase.h>
+#include <ripple/protocol/STIssue.h>
+
+namespace ripple {
+
+class Serializer;
+class STObject;
+
+class STXChainBridge final : public STBase
+{
+    STAccount lockingChainDoor_{sfLockingChainDoor};
+    STIssue lockingChainIssue_{sfLockingChainIssue};
+    STAccount issuingChainDoor_{sfIssuingChainDoor};
+    STIssue issuingChainIssue_{sfIssuingChainIssue};
+
+public:
+    using value_type = STXChainBridge;
+
+    enum class ChainType { locking, issuing };
+
+    static ChainType
+    otherChain(ChainType ct);
+
+    static ChainType
+    srcChain(bool wasLockingChainSend);
+
+    static ChainType
+    dstChain(bool wasLockingChainSend);
+
+    STXChainBridge();
+
+    explicit STXChainBridge(SField const& name);
+
+    STXChainBridge(STXChainBridge const& rhs) = default;
+
+    STXChainBridge(STObject const& o);
+
+    STXChainBridge(
+        AccountID const& srcChainDoor,
+        Issue const& srcChainIssue,
+        AccountID const& dstChainDoor,
+        Issue const& dstChainIssue);
+
+    explicit STXChainBridge(Json::Value const& v);
+
+    explicit STXChainBridge(SField const& name, Json::Value const& v);
+
+    explicit STXChainBridge(SerialIter& sit, SField const& name);
+
+    STXChainBridge&
+    operator=(STXChainBridge const& rhs) = default;
+
+    STObject
+    toSTObject() const;
+
+    AccountID const&
+    lockingChainDoor() const;
+
+    Issue const&
+    lockingChainIssue() const;
+
+    AccountID const&
+    issuingChainDoor() const;
+
+    Issue const&
+    issuingChainIssue() const;
+
+    AccountID const&
+    door(ChainType ct) const;
+
+    Issue const&
+    issue(ChainType ct) const;
+
+    SerializedTypeID
+    getSType() const override;
+
+    Json::Value getJson(JsonOptions) const override;
```

Comment:
Just FYI, if I call `getFullText()` on an `STObject` that contains an `STXChainBridge` I get a malformed result.  Here's a (hand-formatted) example that I got out of some metatdata:
```
            FinalFields =
            {
                Flags = 0,
                OwnerNode = 0,
                XChainClaimID = 1,
                XChainAccountCreateCount = 0,
                XChainAccountClaimCount = 0,
                1000000/XRP,
                200000000/XRP,
                Account = rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh,
                XChainBridge = }
            },
```
My guess is that you need to override the `getText()` and `getFullText()` methods that are inherited from `STBase`.  But I haven't researched the problem any further than simply observing it.

Reply:
Fixed in cb0a8ff11c [fold] Add `getText` to STBridge